app/main.py

```diff
@@ -1,3 +1,62 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
+
+    def __iadd__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            return NotImplemented
+        return self
+
+    def __mul__(self, factor: "int | float") -> "Distance":
+        if isinstance(factor, (int, float)):
+            return Distance(self.km * factor)
+        return NotImplemented
+
+    def __truediv__(self, divisor: "int | float") -> "Distance":
```

Comment:
The type hint `int | float` is only valid in Python 3.10 and later. For compatibility with earlier versions, use `Union[int, float]`.